### PR TITLE
Adapt the Bottom Sheet for Landscape

### LINKED ISSUE #13801: Adapt the Bottom Sheet for Landscape

### PR DESCRIPTION
**Pull request by leandroalonso**
Fixes #13801 

### To test

1. Open the Nudges View
2. Go landscape

Also, open the Nudges view while in Landscape and alternate between Portrait/Landscape.

### Technical detail

The landscape code comes from @bjtitus `BottomSheetPresentationController`, all credits to him. :)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift [hunk 1]
```diff
@@ -21,13 +21,27 @@ public enum DrawerHeight {
...
+
 public protocol DrawerPresentable: AnyObject {
     /// The height of the drawer when it's in the expanded position
     var expandedHeight: DrawerHeight { get }
 
     /// The height of the drawer when it's in the collapsed position
     var collapsedHeight: DrawerHeight { get }
 
+    /// The width of the Drawer in compact screen
+    var width: DrawerWidth { get }
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift:43]: I am worried about this being too vague and could cause conflicts. What about prefixing it? 

`drawerWidth` ?

File: WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift [hunk 2]
```diff
@@ -113,10 +132,36 @@ public class DrawerPresentationController: FancyAlertPresentationController {
...
+                width = givenWidth
+            case .maxWidth:
+                break
+            }
+        }
+
+        /// If we constrain the width, this centers the view by applying the appropriate insets based on width
+        let leftInset: CGFloat = ((containerView.bounds.width - width) / 2)
+
+        return CGRect(x: leftInset, y: frame.origin.y, width: width, height: frame.height)
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift:156]: I don't think we need to create a new CGRect here. We can just modify the values of `frame` directly.

```
/// If we constrain the width, this centers the view by applying the appropriate insets based on width
frame.origin.x = ((containerView.bounds.width - width) / 2)
frame.size.width = width

return frame
```

File: WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift]: Done!

File: WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift]: Done in f666e7c1888220008100034ac53c287bc19c598e

**Comment by leandroalonso** [comment]
@emilylaguna ready for another 👀

**Pull request merged**